src/palace/manager/metadata_layer/container.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from pydantic import BaseModel, ConfigDict
+from sqlalchemy.orm import Session
+
+from palace.manager.metadata_layer.identifier import IdentifierData
+from palace.manager.sqlalchemy.model.datasource import DataSource
+from palace.manager.sqlalchemy.model.identifier import Identifier
+from palace.manager.util.log import LoggerMixin
+
+
+class BaseDataContainer(BaseModel, LoggerMixin):
```

Comment:
This is the mutable base model for `CirculationData` and `Metadata`. I had a bit of trouble coming up with a name for it, and I'm not 100% happy with `BaseDataContainer`, but its the best I could come up with.

Happy to change it if anyone has suggestions.

Reply:
Those are all good suggestions @dbernstein. 

I agree untangling `CirculationData` from `Metadata`, would make things easier to reason about. I actually took a stab at doing that as part of this PR, but it will require some reasonably deep changes, so I ended up rolling it back and decided to defer it to another PR. Once this goes in, I'll see what I can do for a follow up to separate them.

I'll push a commit to this PR to rename `BaseDataContainer` -> `BaseMutableData` and `Metadata` -> `BibliographicData`.

Reply:
I spent some time on this, and I pushed a commit to rename `BaseDataContainer` -> `BaseMutableData` but the `Metadata` rename is going to be much more work. I'm going to merge this one as is, and make a follow up PR with all the changes that are required for the metadata rename.

Reply:
Here's the follow up PR with the rest of the renaming if you want to take a look @dbernstein https://github.com/ThePalaceProject/circulation/pull/2463